keg_elements/db/utils.py

```diff
@@ -82,3 +82,51 @@ def session_flush():
     except Exception:
         db.session.rollback()
         raise
+
+
+class CollectionUpdater(object):
+
+    def __init__(self, entity, attr_name, data):
+        self.entity = entity
+        self.attr_name = attr_name
+        self.data = data
+        self.collection = getattr(entity, attr_name)
+        self.keep_children = set()
+
+        ent_cls = entity.__class__
+        queryable_attr = getattr(ent_cls, attr_name)
+        # Unless they sent in the wrong type, rel_prop should be a
+        # sqlalchemy.orm.properties.RelationshipProperty instance
+        rel_prop = queryable_attr.property
+        # now that we have the property, go through the mapper to get to the child entity class
+        self.child_cls = rel_prop.mapper.class_
+
+    def find_child(self, pk_value):
+        for child in self.collection:
+            if child.id == pk_value:
+                return child
+
+    def update(self):
+        for record in self.data:
+            child = None
+            pk_value = record.get('id')
+            if pk_value:
+                child = self.find_child(pk_value)
+            if not child:
+                # Either not an edit (pk_value is None) or child to edit not found.
+                # Todo: If an edit is requested but no child found, should that be an error?
+                child = self.child_cls(**record)
```

Comment:
should we use `self.child_cls.add()` here so we call potential overrides of the add() class method?